src/components/Preview/Preview.tsx

```diff
@@ -134,20 +136,22 @@ export const Preview = ({
     hmsActions.preview(config);
   }, [config.authToken]);
 
-  useEffect(() => {
-    // @ts-ignore
-    hmsActions.setVideoSettings({ deviceId: selectedVideoInput });
-    // @ts-ignore
-    hmsActions.setAudioSettings({ deviceId: selectedAudioInput });
-  }, [selectedAudioInput, selectedVideoInput]);
-
-  const handleDeviceChange = useCallback((values: SettingsFormProps) => {
-    values?.selectedAudioInput &&
-      setSelectedAudioInput(values.selectedAudioInput);
-    values?.selectedVideoInput &&
-      setSelectedVideoInput(values.selectedVideoInput);
+  const handleDeviceChange = (values: SettingsFormProps) => {
+    const {
+      selectedVideoInput: newSelectedVideoInput,
+      selectedAudioInput: newSelectedAudioInput,
+    } = values;
+    if (newSelectedAudioInput && audioInputDeviceId !== newSelectedAudioInput) {
+      // @ts-ignore
+      hmsActions.setAudioSettings({ deviceId: newSelectedAudioInput });
```

Comment:
should we instead change the type in setAudio/VideoSettings to take Partial, so this ts-ignore is not needed?

Reply:
Yeah we should do that. But that involves change in the store and updating the version in this repo. I was waiting for the opportunity to change the store. Will probably do that as a part of on-role-change-request.